### PR TITLE
adding corrections for forms loading state

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- refactoring how Js init works

### CODE REVIEW

**Review by gabriel-glo — APPROVED** [review]

File: src/Blocks/components/form/assets/index.js [hunk 1]
```diff
@@ -151,14 +135,54 @@ function initAll(manual) {
...
+		// You can disable auto init from the admin.
+		const disableAutoInit = Boolean(esFormsLocalization.formDisableAutoInit) ?? false;
+
+		// Load window form no matter what the option is set.
+		window['esForms'] = {
+			...window['esForms'],
+			initAll: () => {
+				// Bailout if form is loaded but you want to init form again.
+				if (!disableAutoInit) {
+					throw 'You are trying to re-init form class that all-ready exists. Please review your code or disable auto-initialize scripts in the forms global settings.';
```
Comment (iobrado) [inline src/Blocks/components/form/assets/index.js:164]:
```suggestion
					throw 'You are trying to re-init form class that already exists. Please review your code or disable auto-initialize scripts in the forms global settings.';
```

**Review by iobrado — APPROVED** [review]
Nice! Found just one typo 🙂

**Review by mbmjertan — APPROVED** [review]
Looks good, @iobrado was faster 😄

File: src/Hooks/Form.md [hunk 2]
```diff
@@ -46,49 +62,70 @@ Here are events that you can trigger using JavaScript to hook you custom logic i
...
+		events,
+	} = window.esForms;
+
+	window.addEventListener(
+		events.FORMS_JS_LOADED,
+		() => {
+			const form = document.querySelector(formSelector);
+
+			form?.addEventListener('esFormsAfterFormSubmit', (event) => {
+				// Do you logic here.
```
Comment (volfkarlo) [inline src/Hooks/Form.md:85]:
```suggestion
				// Do your magic! ✨ 
```
or so

File: src/Hooks/Form.md [hunk 3]
```diff
@@ -17,17 +17,33 @@ window.esForms.init();
 This will manually re-init all JS for the forms, you can use this option in combination with the global setting to disable auto-init of JS on page load.
 
 ```js
-window.addEventListener('load', function () {
+function initForms() {
 	if (!window?.esForms) {
```
Comment (volfkarlo) [inline src/Hooks/Form.md:21]: Can we guarantee that our `esForms` property is available in the window object before we come to this point?
(Plugin code before our theme code?)

**Review by volfkarlo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**